client/src/utils/uploadImage.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import { Button } from "../components/Button";
+import UploadPreview from "../components/UploadPreview";
+
+export default function UploadImage() {
+  const [uploadInput, setUploadInput] = useState("");
+  const [previewSrc, setPreviewSrc] = useState("");
+
+  const handleImageChange = (event) => {
+    const file = event.target.files[0];
+    console.log("file", file);
+    previewFile(file);
+    setUploadInput(event.target.value);
+  };
+
+  const previewFile = (file) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onloadend = () => {
+      setPreviewSrc(reader.result);
+      console.log(reader.result);
+    };
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!previewSrc) {
+      console.log("No preview file");
+    }
+    uploadImage(previewSrc);
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await fetch("/api/upload", {
+        method: "POST",
+        body: JSON.stringify({ data: base64EncodedImage }),
+        headers: { "Content-Type": "application/json" },
+      });
```

Comment:
This would be great as a utils file

Reply:
Would that be the async function as a seperate file in the utils folder, imported into the UploadImage file?